system/lib/libcxxabi/src/cxa_personality.cpp

```diff
@@ -964,6 +966,11 @@ __gxx_personality_v0
             exc->languageSpecificData = results.languageSpecificData;
             exc->catchTemp = reinterpret_cast<void*>(results.landingPad);
             exc->adjustedPtr = results.adjustedPtr;
+#ifdef __USING_WASM_EXCEPTIONS__
+            // Wasm only uses a single phase (_UA_SEARCH_PHASE), so save the
+            // results here.
+            set_registers(unwind_exception, context, results);
+#endif
```

Comment:
This is to save our selector computation result.

In the upstream version, the search phase searches for a matching catch handler and caches the result, and the second cleanup phase retrieves the result and sets registers with the final selector value. (In Wasm's libunwind, they are not registers but just a field in a struct.). They do it in the second cleanup phase: https://github.com/emscripten-core/emscripten/blob/40fb7d2071e439f1de614898b88518df582faa94/system/lib/libcxxabi/src/cxa_personality.cpp#L973

But now we only use the search phase, we do it here. Instead, we are able to remove our duplicated caching code in line 984-993 below.